### ISSUE TITLE
some sound files won't play [bug]

### ISSUE THREAD

**Issue by LocalGod79** [endless-sky/endless-sky#1575]
http://steamcommunity.com/app/404410/discussions/1/350542114327534758/

**Comment by endless-sky** [comment]
Given that #1515 is specifically about the zapper noise being annoying, I doubt it's as simple as "z" sounds not playing (more likely is that it might be happening with the very last sound in alphabetical order). One person in the steam thread says that making a new pilot "fixed" it, and another says that it only happened when they were using plugins.

**Comment by LocalGod79** [comment]
i'll admit i read it quickly, but @Amazinite suggested that this required a github issue, so i obliged.

**Comment by Amazinite** [comment]
It's something I'd notice on a few rare occasions (sometimes fixing itself after a few play sessions), but having seen someone else have the issue I thought it deserved github attention.

**Comment by Amazinite** [comment]
It appears that if any plugins are installed, the last sound at the end of every single sounds file will no be playable.

**Comment by endless-sky** [comment]
I've tested this by creating a plugin with sounds, and the sounds play just fine - including the last one in the folder. So, I suspect that something else was going on here, like a sound file that was the wrong format.

**Comment by Amazinite** [comment]
It must be an OS thing then, because I just did the same thing on Win 8 and the last sound in the folder will never play. I have to make an extra sound and name it "zzz" or something in order to make the sound that was last play.

**Comment by Bladewood** [comment]
I installed plugins.
Zapper sound did not work.

I made a dummy sound file, `z_dummy sound.mp3`, and placed it in my sounds folder.
Sound worked.
Removed that file.
Sound did not work.

Removed plugins from my plugins folder.
Sound worked.

Installed plugin containing two custom sound files.
Alphabetically last sound of that plugin did not work.

Windows 10.

**Comment by endless-sky** [comment]
I can't reproduce this on Windows 7 - I tried creating a plugin, and the sounds all loaded just fine. So unless one of you who are able to reproduce the bug can track down what's causing it, I can't make any more progress on trying to fix it.

**Comment by endless-sky** [comment]
You could also test and see if commit 9680dcf either fixes this, or at least produces an error message saying that the file could not be loaded.

**Comment by Amazinite** [comment]
Yup, it's fixed now. Plugin sounds that use to not play are now playing.

**Comment by Bladewood** [comment]
YAY IT WORKS

**Comment by comnom** [comment]
@endless-sky This bug still exists in a strange form. I tried to add a sound to my plugin, spit.wav, and got an error that it could not be loaded.

The sound was in plugins/Eternals/sounds/spit.wav, but the error message was for plugins/eternals-high-dpi/sounds/spit.wav. There is no sound folder in that plugin at all.

Eventually I renamed it to try and get it working, and found out the following:
If a sound is the last sound in the plugin folder, and if that sound's name is less than or equal to five characters long, it throws an error.

I found this out by naming the file something.wav, which loads fine, and renaming it with one less character each run until it threw the following: 
Unable to load sound "somet" from path: C:/Users/WIN7HP/AppData/Roaming/endless-sky/plugins/eternals-high-dpi/sounds/somet.wav

The only other thing I can offer is that eternals-high-dpi is the last plugin to load(but it shouldn't be trying to load sounds from there anyways).
Win 7 x64, build is only a day or two old.
